apps/api/src/common/domainFunctions/registry.ts

```diff
@@ -0,0 +1,21 @@
+import {
+  DomainHandlerMap,
+  GenericHandler,
+  ResolvedDomainHandlerMap,
+} from '@api/domain';
+import {getHandlerMetadata} from './handlers';
+
+export class HandlerRegistry<T extends object = DomainHandlerMap> {
+  #events = new Map<string, GenericHandler>();
+
+  get handlers(): ResolvedDomainHandlerMap<T> {
+    return Object.fromEntries(
+      this.#events.entries(),
+    ) as unknown as ResolvedDomainHandlerMap<T>;
```

Comment:
@riccardoperra what do you think instead cast to unknown if we create a function to get typed entries?
something like this:

```ts
declare type Entries<T> = {
  [K in keyof T]: [K, T[K]];
}[keyof T][];

export const entries = <
  T extends Record<string, unknown> | Map<string, unknown>,
>(
  obj: T,
): Entries<T> => Object.entries(obj) as any;
```

Reply:
I'm not sure this works because the #entries type is Map<string, GenericHandler>, we need to force the cast because it should resolve type based on "T", assuming that you are registering ALL handlers you are defining

Reply:
@hackpirodev I also noticed that the current implementation doesn't need `as unknown`, so it's basically the same. we can remove it using the same fromEntries() fn